TinyMCE 4.9.11 does not build

**What is the current behavior? Describe the bug**

```sh
$ git checkout 4.9.11
$ grunt
Running "shell:tsc" (shell) task
src/core/test/ts/browser/bookmark/BookmarksTest.ts(33,30): error TS2345: Argument of type '(input: Record<string, any>) => import(".../tinymce/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<Record<string, any>, any>' is not assignable to parameter of type '(input: Record<string, any>) => import(".../tinymce/node_modules/@ephox/agar/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<Record<string, any>, any>'.
  Call signature return types 'Result<Record<string, any>, any>' and 'Result<Record<string, any>, any>' are incompatible.
    The types of 'map(...).mapError(...).bind(...).map' are incompatible between these types.
      Type '<U>(mapper: (value: unknown) => U) => Result<U, U>' is not assignable to type '<U>(mapper: (value: U) => U) => Result<U, U>'.
        Types of parameters 'mapper' and 'mapper' are incompatible.
          Types of parameters 'value' and 'value' are incompatible.
            Type 'unknown' is not assignable to type 'U'.
              'U' could be instantiated with an arbitrary type which could be unrelated to 'unknown'.
src/core/test/ts/browser/selection/SimpleTableModelTest.ts(18,25): error TS2345: Argument of type '(_: unknown) => Result<unknown, string>' is not assignable to parameter of type '(input: unknown) => Result<unknown, string>'.
  Call signature return types 'Result<unknown, string>' and 'Result<unknown, string>' are incompatible.
    The types returned by 'map(...).mapError(...)' are incompatible between these types.
      Type 'import(".../tinymce/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<U, U>' is not assignable to type 'import(".../tinymce/node_modules/@ephox/agar/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<U, U>'.
src/plugins/imagetools/test/ts/module/test/ImageOps.ts(11,25): error TS2345: Argument of type '(element: unknown) => Result<unknown, any>' is not assignable to parameter of type '(input: unknown) => Result<unknown, any>'.
  Call signature return types 'Result<unknown, any>' and 'Result<unknown, any>' are incompatible.
    The types returned by 'map(...)' are incompatible between these types.
      Type 'import(".../tinymce/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<U, any>' is not assignable to type 'import(".../tinymce/node_modules/@ephox/agar/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<U, any>'.
src/themes/mobile/test/ts/module/test/ui/TestUi.ts(6,34): error TS2345: Argument of type '() => Result<unknown, string>' is not assignable to parameter of type '(input: unknown) => Result<unknown, string>'.
  Type 'import(".../tinymce/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<unknown, string>' is not assignable to type 'import(".../tinymce/node_modules/@ephox/agar/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<unknown, string>'.
src/themes/mobile/test/ts/module/test/ui/TestUi.ts(12,33): error TS2345: Argument of type '(elem: Element<any>) => Result<unknown, string>' is not assignable to parameter of type '(input: Element<any>) => Result<unknown, string>'.
  Type 'import(".../tinymce/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<unknown, string>' is not assignable to type 'import(".../tinymce/node_modules/@ephox/agar/node_modules/@ephox/katamari/lib/main/ts/ephox/katamari/api/Result").Result<unknown, string>'.
Warning: Command failed: node ./node_modules/typescript/bin/tsc
 Use --force to continue.

Aborted due to warnings.
```

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**

See above.

**What is the expected behavior?**

TinyMCE builds normally.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

Debian 9, Node 10.22.0. Not sure if this works in previous versions of TinyMCE, I specifically want to build the latest 4.x version.

--- a/readme.md
+++ b/readme.md
@@ -12,13 +12,13 @@ Open a console and go to the project directory.
 ```
 $ cd tinymce/
 ```
-Install `grunt` command line tool globally.
+Install the `yarn` and `grunt` command line tools globally.
 ```
-$ npm i -g grunt-cli
+$ npm i -g yarn grunt-cli
 ```
 Install all package dependencies.
 ```
-$ npm install
+$ yarn
 ```
 Now, build TinyMCE by using `grunt`.
 ```

